File: meldingsvarsler-fasit-forslag/src/main/java/no/sonat/meldingsvarsler/epost/EpostProsessor.java
package no.sonat.meldingsvarsler.epost;

import no.sonat.meldingsvarsler.abonnent.Abonnent;
import no.sonat.meldingsvarsler.Melding;
import no.sonat.meldingsvarsler.MeldingProsessor;

public class EpostProsessor implements MeldingProsessor {
    @Override
    public boolean haandtererUtsendingAv(Melding melding) {
        return melding instanceof EpostMelding;
    }

    @Override
    public void sendMelding(Abonnent abonnent, Melding melding) {
        sendEpost(((EpostAbonnent) abonnent), (EpostMelding) melding);
    }
    public void sendEpost(EpostAbonnent epostAbonnent, EpostMelding epostMelding) {
        System.out.println("Sender Epost til "
                + epostAbonnent.epostAdresse()
                + "(" + epostAbonnent.navn() + ")"
                + " => " + epostMelding.epostTittel()
                + " => " + epostMelding.epostMelding());
    }
}

File: meldingsvarsler-fasit-forslag/App/EpostProsessor.cs
using System;

namespace App
{
    public class EpostProsessor : IMeldingProsessor
    {
        public bool HaandtererUtsendingAv(IMelding melding)
        {
            return melding is EpostMelding;
        }

        public void SendMelding(IAbonnent abonnent, IMelding melding)
        {
            SendEpost((EpostAbonnent) abonnent, (EpostMelding) melding);
        }

        public void SendEpost(EpostAbonnent epostAbonnent, EpostMelding epostMelding)
        {
            Console.Out.WriteLine("Sender Epost til "
                                + epostAbonnent.EpostAdresse
                                + "(" + epostAbonnent.Navn + ")"
                                + " => " + epostMelding.Tittel
                                + " => " + epostMelding.Melding);
        }
    }
}